src/Orchard.Environment.Shell.Data/ShellDescriptorManager.cs

```diff
@@ -31,14 +32,20 @@ public class ShellDescriptorManager : IShellDescriptorManager
             _eventBus = eventBus;
             _session = session;
             _logger = logger;
+
+            _asyncLazyShellDescriptor = new Lazy<Task<ShellDescriptor>>(() => Task.Run(() =>
```

Comment:
We can't run a precompilation with some db access.
If we need to change the way the shape tag helpers are done, I will do it.